pddl/parser/domain.py

```diff
@@ -86,9 +86,11 @@ def requirements(self, args):
 
     def types(self, args):
         """Parse the 'types' rule."""
-        if not bool({Requirements.TYPING} & self._extended_requirements):
+        if any(args[2].values()) and not bool(
```

Comment:
`args[2].values()`??

Reply:
While following the BNF, I noticed that the `:typing` requirement is needed only when a hierarchy is defined, i.e., when the type_sep (`-`) is used. This is actually a tiny detail, we can get rid of it or I may try to make it more readable.